src/components/SettlementButton.js

```diff
@@ -88,43 +106,64 @@ const defaultProps = {
     },
 };
 
-class SettlementButton extends React.Component {
-    componentDidMount() {
+function SettlementButton({
+    addDebitCardRoute,
+    addBankAccountRoute,
+    anchorAlignment,
+    betas,
+    buttonSize,
+    chatReportID,
+    currency,
+    enablePaymentsRoute,
+    iouReport,
+    isDisabled,
+    isLoading,
+    formattedAmount,
+    nvp_lastPaymentMethod,
+    onPress,
+    policyID,
+    shouldShowPaymentOptions,
+    shouldShowPaypal,
+    style,
+}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
         PaymentMethods.openWalletPage();
-    }
+    }, []);
 
-    getButtonOptionsFromProps() {
+    const getButtonOptionsFromProps = useMemo(() => {
```

Comment:
Update the name, isn't a function now. It should be something like `paymentOptions` or `paymentButtonOptions`